src/Stratis.Bitcoin.Features.Miner/PowMining.cs

```diff
@@ -310,12 +310,12 @@ public int IncrementExtraNonce(Block block, ChainedHeader previousHeader, int ex
             }
 
             extraNonce++;
+
             int height = previousHeader.Height + 1; // Height first in coinbase required for block.version=2
-            Transaction txCoinbase = block.Transactions[0];
-            txCoinbase.Inputs[0] = TxIn.CreateCoinbase(height);
 
-            Guard.Assert(txCoinbase.Inputs[0].ScriptSig.Length <= 100);
-            block.UpdateMerkleRoot();
+            block.Transactions[0].Inputs[0].ScriptSig = new Script(Op.GetPushOp(height)) + OpcodeType.OP_0; // update TxIn.ScriptSig, do not overwrite it, it can contain a WitScript
```

Comment:
I think this comment is a bit confusing if you're unfamiliar with what `TxIn.WitScript` actually is.

```suggestion
            block.Transactions[0].Inputs[0].ScriptSig = new Script(Op.GetPushOp(height)) + OpcodeType.OP_0; // We only want to update the ScriptSig, without overwriting the entire TxIn, as the TxIn can contain witness data in `TxIn.WitScript`.
```

Reply:
yeah its a bit hacky, I will see if I can change it